cpp/include/cudf/quantiles.hpp

```diff
@@ -42,5 +42,32 @@ quantile(column_view const& input,
          interpolation interp = interpolation::LINEAR,
          order_info column_order = {});
 
+/* @brief Gathers the rows corrosponding to the desired quantiles. The indices
+ *        used to gather rows are computed by interpolating between the index on
+ *        either side of the desired quantile. Since some columns may be non-
+ *        arithmetic, interpolation between rows is limited to non-arithmetic
+ *        strategies.
+ *
+ * @param input           Table used to compute quantile rows.
+ * @param q               Desired quantiles in range [0, 1].
+ * @param interp          Strategy used to select between the two rows on either
+                          side of the desired quantile.
+ * @param sorted          Indicates if the input has been pre-sorted.
+ * @param column_order    The desired sort order for each column.
+ * @param null_precedence The desired order of null compared to other elements.
+ *
+ * @throws cudf::logic_error if `interp` is an arithmetic interpolation strategy
+ * @throws cudf::logic_error if `input` is empty
+ */
+std::unique_ptr<table>
+quantiles(table_view const& input,
+          std::vector<double> const& q,
+          interpolation interp = interpolation::NEAREST,
+          cudf::sortedness sorted = sortedness::UNSORTED,
```

Comment:
I think @codereport 's advice here would be that predicates should be prefixed with `is_/has_`. 
```suggestion
          cudf::sortedness is_sorted = sortedness::UNSORTED,
```

Reply:
100%, see reply from Guy Davidson to one of my tweets: https://twitter.com/hatcat01/status/1007161606994845697?s=20

Two follow up points:
1. It is widely acknowledged that the `empty` method on most STL data structures was a mistake because we don't know if it means `clear` (as a verb) or `is_empty` (a predicate aka function that returns `true` or `false`)
2. In python, `sorted` is a free function that sorts a list, for example:
```py
x = y = [3, 2, 1]
x.sort()
print(x == sorted(y)) # prints: True
``` 
so we should definitely call this `is_sorted`

All that being said, I just realized this is not a `predicate` function, it is a parameter. I still think `is_sorted` is better though. A recommendation on top of that would be to change:
```cpp
enum class sortedness : bool {
   SORTED,
   UNSORTED
};
```
to 
```cpp
enum class sorted : bool {
   NO, 
   YES
};
```
The comparison code then goes from:
```cpp
if (sorted == sortedness::SORTED) { ... }
```
to 
```cpp
if (sorted == sorted::YES) { .. }
```
Reduce the number of times `sort` shows up by 1 :)

Reply:
```
enum class sorted : bool {
   NO, 
   YES
};
```

I like this suggestion.

Reply:
> > The comparison code then goes from:
> 
> if (sorted == sortedness::SORTED) { ... }
> to
> 
> if (sorted == sorted::YES) { .. }

I assume you meant:

`if(is_sorted == sorted::YES)`

Reply:
How do you feel about this?

```
switch (is_sorted) {
sorted::YES:
    // we know we don't need to sort.
sorted::NO:
    // we know we need to sort.
sorted::UNKNOWN:
    // we don't know if we need to sort, try `is_sorted` first?
}
```

Reply:
`is_sorted` is probably expensive. I think just keeping it a binary choice is good.

Reply:
Also, the reason I was using `sortedness` and `sorted` is because `is_sorted` is already a `cudf` function.

Reply:
@jrhemstad github wasn't pushing me comments, so I ended up editing my comment after you posted. Sorry.

Reply:
I just realized that `cudf::is_sorted(...)`(the function) would conflict with the `enum class` proposed here.

Reply:
`enum class sortedness` and `sortedness sorted` solve that problem, but feel weird. `enum class sorted` and `sorted is_input_sorted` would alleviate it, though?

Reply:
Updated.